src/locale/keymap-util.c

```diff
@@ -780,3 +781,137 @@ int x11_convert_to_vconsole(Context *c) {
+        }
+
+        first_line = true;
+        for (;;) {
+                _cleanup_free_ char *line = NULL;
+                char *line_locale;
+
+                r = read_line(fr, LONG_LINE_MAX, &line);
+                if (r < 0)
+                        return r;
+                if (r == 0)
+                        break;
+
+                if (locale_enabled) {
+                        /* Just complete writing the file if the new locale was already enabled */
+                        fprintf(fw, "%s%s", first_line? "" : "\n", line);
+                        first_line = false;
+                        continue;
+                }
+
+                line = strstrip(line);
+                if (isempty(line)) {
+                        fputs("\n", fw);
+                        first_line = false;
+                        continue;
+                }
+
+                line_locale = line;
+                if (line_locale[0] == '#')
+                        line_locale = strstrip(line_locale + 1);
+
+                if (strcasecmp_ptr(line_locale, locale_entry) == 0) {
+                        /* Uncomment existing line for new locale */
+                        fprintf(fw, "%s%s", first_line? "" : "\n", locale_entry);
+                        locale_enabled = true;
+                        first_line = false;
+                        continue;
+                }
+
+                /* The line was not for the locale we want to enable, just copy it */
+                fprintf(fw, "%s%s", first_line? "" : "\n", line);
+                first_line = false;
+        }
+
+        if (!locale_enabled) {
+                /* The locale to enable was not found as commented line, add it to the end */
+                fprintf(fw, "\n%s", locale_entry);
+        }
+        fputs("\n", fw);
+
+        r = fflush_sync_and_check(fw);
+        if (r < 0)
+                goto fail;
+
+        if (rename(temp_path, LOCALE_GEN_CONF_FNAME) < 0) {
+                r = -errno;
+                goto fail;
+        }
+
+        return 0;
+
+fail:
+        if (temp_path)
+                (void) unlink(temp_path);
+        return r;
+}
+
+int locale_gen_run(void) {
+        struct stat st;
+        pid_t pid;
+        int r;
+        const char *LOCALEGEN_BIN = "/usr/sbin/locale-gen";
+
+        r = stat(LOCALEGEN_BIN, &st);
+        if (errno == ENOENT)
+                return 0;
+        if (r < 0)
+                return -errno;
+
+        r = safe_fork("(sd-localegen)", FORK_RESET_SIGNALS|FORK_RLIMIT_NOFILE_SAFE|FORK_CLOSE_ALL_FDS|FORK_DEATHSIG|FORK_LOG, &pid);
+        if (r < 0)
+                return r;
+        if (r == 0) {
+                execl(LOCALEGEN_BIN, LOCALEGEN_BIN, NULL);
+                log_open();
+                log_error_errno(errno, "Failed to invoke %s: %m", LOCALEGEN_BIN);
+                _exit(EXIT_FAILURE);
+        }
+
+        r = wait_for_terminate_and_check(LOCALEGEN_BIN, pid, WAIT_LOG_ABNORMAL);
+        return r;
```

Comment:
please ifdef this all under some new conditoin so that non-debian distros can compile this out.